IOT/main/scheduler.h

```diff
@@ -0,0 +1,28 @@
+#ifndef SCHEDULER_H
+#define SCHEDULER_H
+
+#define MAX_TASKS 20
+
+struct task_t {
+  // Delay before the task is executed (real execution delay may vary)
+  unsigned long delay;
+  // Time when the task was last executed
+  unsigned long lastRun;
+  // Function must be called indefinitely
+  bool loop;
+  // Function to be executed
+  void (*task)();
```

Comment:
You might want to add a `void *` context to this function pointer (as well as storing said pointer in task_t)

An example is worth 1000 words:
```cpp
struct task_t {
  // [...]
  void (*task)(void *context);
  void *context;
}

struct SomeContext {
  int a;
  char *b;
}

void reentrant_task(void *context) {
  SomeContext *c = reinterpret_cast<SomeContext *>(context);
  c->a++;
  Serial.print("This task was run ");
  Serial.print(c->a);
  Serial.println("times!");
}

int main() {
  SomeContext *c = new SomeContext { 0, "yes" };
  task_t task = { 0UL, 0UL, false, &reentrant_task, c };
  scheduler s;
  s.addTask(task);
  s.run();
  delete c;
}
```